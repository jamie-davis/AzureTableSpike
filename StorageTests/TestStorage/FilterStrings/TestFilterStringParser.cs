using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using StorageTests.TestUtilities;
using TestConsoleLib;
using TestStorage.FilterStrings;

namespace StorageTests.TestStorage.FilterStrings
{
    [TestFixture]
    [UseReporter(typeof(CustomReporter))]
    public class TestFilterStringParser
    {
        [Test]
        public void FilterStringsAreParsed()
        {
            //Arrange
            var output = new Output();
            var strings = new[]
                {
                    "var eq 1",
                    "var eq 1 and var2 eq 'thing'",
                    "(var eq 1)",
                    "(var eq 1 or var2 eq 'thing') and var3 gt 56",
                    "var eq guid'9E37E338-27B2-4F32-AFFA-DC74F017AF1D' or (var2 ge 45L and var3 ne 35.4)",
                    "eq eq eq",
                    "and",
                    "and and and",
                    "(a eq 1",
                    "a eq 1)",
                    "PartitionKey eq '2017-07-30T21:15:09' and (Alpha eq 'Delta' and (Delta eq 'Alpha' or Delta eq 'Beta'))"
                };

            //Act
            foreach (var filter in strings)
            {
                var result = FilterStringParser.Parse(filter);

                output.WrapLine($@"Filter string: {filter}");
                output.WrapLine(result.ToString());
                output.WriteLine();
                output.WriteLine();
            }

            //Assert
            Approvals.Verify(output.Report);
        }
    }
}
