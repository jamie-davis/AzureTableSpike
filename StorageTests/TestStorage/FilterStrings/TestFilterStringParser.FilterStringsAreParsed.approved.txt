Filter string: var eq 1                                                                        
var eq 1                                                                                       
                                                                                               
                                                                                               
Filter string: var eq 1 and var2 eq 'thing'                                                    
var eq 1 and var2 eq thing                                                                     
                                                                                               
                                                                                               
Filter string: (var eq 1)                                                                      
(var eq 1)                                                                                     
                                                                                               
                                                                                               
Filter string: (var eq 1 or var2 eq 'thing') and var3 gt 56                                    
(var eq 1 or var2 eq thing) and var3 gt 56                                                     
                                                                                               
                                                                                               
Filter string: var eq guid'9E37E338-27B2-4F32-AFFA-DC74F017AF1D' or (var2 ge 45L and var3 ne   
35.4)                                                                                          
var eq 9E37E338-27B2-4F32-AFFA-DC74F017AF1D or (var2 ge 45 and var3 ne 35.4)                   
                                                                                               
                                                                                               
Filter string: eq eq eq                                                                        
eq eq eq                                                                                       
                                                                                               
                                                                                               
Filter string: and                                                                             
Error: Unable to parse filter string.                                                          
                                                                                               
                                                                                               
Filter string: and and and                                                                     
Error: Unable to parse filter string.                                                          
                                                                                               
                                                                                               
Filter string: (a eq 1                                                                         
Error: Unable to parse filter string.                                                          
                                                                                               
                                                                                               
Filter string: a eq 1)                                                                         
Error: Unable to parse filter string.                                                          
                                                                                               
                                                                                               
Filter string: PartitionKey eq '2017-07-30T21:15:09' and (Alpha eq 'Delta' and (Delta eq       
'Alpha' or Delta eq 'Beta'))                                                                   
PartitionKey eq 2017-07-30T21:15:09 and (Alpha eq Delta and (Delta eq Alpha or Delta eq Beta)) 
                                                                                               
                                                                                               
                                                                                               